addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/ChannelTypeUtils.java

```diff
@@ -13,26 +13,43 @@
 
 import org.eclipse.smarthome.core.library.types.DateTimeType;
 import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
 
 /**
  * This class holds various channel values conversion methods
  *
  * @author Gaël L'hopital
  */
 public class ChannelTypeUtils {
-    public static DateTimeType toDateTimeType(Integer netatmoTS) {
+    public static State toStringType(String value) {
+        return (value == null) ? UnDefType.NULL : new StringType(value);
+    }
+
+    public static State toDateTimeType(Integer netatmoTS) {
         Calendar calendar = Calendar.getInstance();
         calendar.setTimeInMillis(netatmoTS * 1000L);
-        return new DateTimeType(calendar);
+        return toDateTimeType(calendar);
+    }
+
+    public static State toDateTimeType(Calendar calendar) {
+        return (calendar == null) ? UnDefType.NULL : new DateTimeType(calendar);
+    }
+
+    public static State toDecimalType(float value) {
+        return toDecimalType(new BigDecimal(value));
     }
 
-    public static DecimalType toDecimalType(float value) {
-        BigDecimal decimal = new BigDecimal(value).setScale(2, BigDecimal.ROUND_HALF_UP);
-        return new DecimalType(decimal);
+    public static State toDecimalType(double value) {
```

Comment:
Should you maybe even accept `Double` and `Float` and add `null` checks in these methods as well. Because according to the interface it returns the object versions and so potential `null`